src/node_worker.cc

```diff
@@ -481,6 +482,7 @@ void Worker::New(const FunctionCallbackInfo<Value>& args) {
     THROW_ERR_MISSING_PLATFORM_FOR_WORKER(env);
     return;
   }
+  CHECK(!env->isolate_data()->is_building_snapshot());
```

Comment:
Would it make sense to write a descriptive message in case this fails using `CHECK_WITH_MSG`?

Reply:
This would only fail if someone removes the `throwIfBuildingSnapshot()` part from JS land, in that case I think they'd already know what they are doing?